package.json

```diff
@@ -32,28 +32,32 @@
     "test:e2e": "jest --config ./test/jest-e2e.json"
   },
   "dependencies": {
-    "@nestjs/common": "^6.0.0",
-    "@nestjs/core": "^6.0.0",
-    "mqtt": "^3.0.0",
-    "reflect-metadata": "^0.1.12",
-    "rimraf": "^2.6.2",
-    "rxjs": "^6.3.3"
+    "mqtt": "4.3.1"
```

Comment:
Actually, using `4.3.1` didn’t solve the issue. I still get `Cannot set property length of #<Map> which has only a getter`. Latest commit that does not contain `collections` dependency is [8aa2f8d8](https://github.com/mqttjs/MQTT.js/commit/8aa2f8d82f4b74706882cd61b36b240e04511b44) (`collections` were added in [c92b877](https://github.com/mqttjs/MQTT.js/commit/c92b877292d314e3e0b5d8f84b7f4b68a266aba2)).

I understand this is not an issue of `nest-mqtt`, but of `mqtt`, but nevertheless it causes issues with other packages utilising `nest-mqtt`/`mqtt` and using the _standard_ Node methods modified by `collections` package.

Reply:
@microud, actually, last version without `collections` dependency is `v4.2.8` and last commit without it is 8aa2f8d, therefore even `4.3.1` does not solve the issue (at least not for me). I wish they would remove `collections` from `mqtt` dependencies.

Reply:
@tukusejssirs I notice `mqtt.js` has released version `v4.3.3` which removed collections.js dependency. I will upgrade dependence.

I no longer use mqtt for a long while and I don't have a complete environment for logical testing. I will upgrade dependance and publish a new version and it's great if you could help me test. Thanks!

Reply:
> @tukusejssirs I notice `mqtt.js` has released version `v4.3.3` which removed collections.js dependency. I will upgrade dependence.

No, it wasn’t removed yet, just its use diminished. I already [tested](https://github.com/mqttjs/MQTT.js/issues/1392#issuecomment-1006026205) it, but still cannot use it.

> I could remove mqtt dependence and let user decide which version should be used.

That could be a way to deal it. Even `@nestjs/microservices` has set [`mqtt` version to `*`](https://github.com/nestjs/nest/blob/master/packages/microservices/package.json#L37).

Reply:
There’s a new PR which removed the `collections` dependency: mqttjs/MQTT.js#1396. It should fix my issues.

However, using `*` would be better IMO.

Reply:
@tukusejssirs Yes, remove lib dependence is the better way for nest wrapped lib. It could break the binding relationship between nest module and base lib.